src/lib/journey/journey.store.ts

```diff
@@ -140,43 +132,51 @@ export function initialize(initOptions?: StepOptions): JourneyStore {
         });
       }
 
-      /** *******************************************************************
-       * SDK INTEGRATION POINT
-       * Summary: Repopulate callbacks/payload with previous data.
-       * --------------------------------------------------------------------
-       * Details: Now that we have a new authId (the identification of the
+      /**
+       * Now that we have a new authId (the identification of the
        * fresh step) let's populate this new step with old callback data if
-       * allowed with this stage. If not, the user will have to refill form. We
-       * will display the error we collected from the previous submission,
-       * restart the flow, and provide better UX with the previous form data,
-       * so the user doesn't have to refill the form.
+       * this is step one and meets a few criteria.
+       *
+       * If error code is 110 or error message includes "Constrained Violation",
+       * then the issue needs special handling.
+       *
+       * If this is the first step in the journey, replace the callbacks with
+       * existing callbacks to resubmit with a fresh authId.
        ******************************************************************* */
-      if (restartedStep.type === StepType.Step) {
+      if (
+        shouldPopulateWithPreviousCallbacks(nextStep, previousCallbacks, restartedStep, stepNumber)
+      ) {
+        /**
+         * TypeScript notes:
+         *
+         * Assert that restartedStep is FRStep as that is required for the above condition to be true.
+         * Also, assert that previousCallbacks is FRCallback[] as that too is required for above to be true.
+         *
+         * Attempt a refactor using Ryan's suggestion found here: https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBZJcWD1RqgJyofGcABqDGg7EgAB4cAA+AAUq3y3nBqwUPGEglQlE4IwA-FcJcNQALOOxENJvgBKUAAb0UJT1CNAPNQ7SJoIAvBbQAAiZWq75WzV0hmgUG6vXg6CCFBOsheVZukoB0CKAC+incNCGUtAZtpkHpvuZrI54slzF5VoAjA6ANzkynUrxCYjyqV8gWphUAH36KrVZHVAuB8BaXh17oNRpNqXNloA5JWpX3Ne33XqfZkyGy8+6w0GJziWV683PO8XS8wjXFmOqR0Go8wAhlYKzuPoeVbsNBoPBc6HgiocM0PL7QIh4H0GMD2JG7owpewDDMJA-AnuoiRfvAegiF4VoAKKrAwiALPoVpJNiVrgA4qADqAABykASFaQQqAA8l8ZDvF6-DAUcqCOAorjSHgcbvjoCpfF6aKINCwiXF8kgftEIgGBw2ILEwrAcDwQQlEAA
```

Comment:
Didn't have time to refactor this TypeScript issue we discussed, but I left this comment, so we can easily revisit it.